fusionauth/resource_fusionauth_application.go

```diff
@@ -686,10 +718,26 @@ func newOAuthConfiguration() *schema.Resource {
 				Default:     false,
 				Description: "When enabled the user will be required to be registered, or complete registration before redirecting to the configured callback in the authorization code grant or the implicit grant. This configuration does not currently apply to any other grant.",
 			},
-			"provided_scope_policy": {
-				Type:     schema.TypeList,
-				Optional: true,
-				Elem:     newOAuthConfigurationScopePolicy(),
+			"scope_handling_policy": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     fusionauth.OAuthScopeHandlingPolicy_Strict.String(),
+				Description: "Controls the policy for handling of OAuth scopes when populating JWTs and the UserInfo response. The possible values are: `Compatibility` - OAuth workflows will populate JWT and UserInfo claims in a manner compatible with versions of FusionAuth before version 1.50.0. `Strict` - OAuth workflows will populate token and UserInfo claims according to the OpenID Connect 1.0 specification based on requested and consented scopes.",
+				ValidateFunc: validation.StringInSlice([]string{
+					fusionauth.OAuthScopeHandlingPolicy_Compatibility.String(),
+					fusionauth.OAuthScopeHandlingPolicy_Strict.String(),
+				}, false),
+			},
+			"unknown_scope_policy": {
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
Let's make this one required as well. The migration set this to `Allow` in case existing applications were relying on unknown scopes being passed through. New applications set this to `Reject` by default.